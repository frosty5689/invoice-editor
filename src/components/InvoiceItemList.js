import React from 'react';
import { connect } from 'react-redux';
import InvoiceItem from './InvoiceItem';
import AddInvoiceItem from './AddInvoiceItem';
import { getLineItems } from '../redux/selectors';

export const InvoiceItemList = ({ lineItems }) => {
  const invoiceItems = lineItems && lineItems.length ? lineItems.map((item, index) => {
    return <InvoiceItem key={item.id} item={item} />;
  }) : null;

  return (
    <tbody>
      {invoiceItems}
      <tr>
        <td><AddInvoiceItem /></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
      </tr>
    </tbody>
  );
}

const mapStateToProps = state => {
  const lineItems = getLineItems(state);
  return {lineItems};
}

export default connect(mapStateToProps)(InvoiceItemList);