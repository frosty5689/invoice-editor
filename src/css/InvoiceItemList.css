.InvoiceItemList-fade-enter {
  opacity: 0.01;
}
.InvoiceItemList-fade-enter-active {
  opacity: 1;
  transition: opacity 500ms ease-in;
}
.InvoiceItemList-fade-exit {
  opacity: 1;
}
.InvoiceItemList-fade-exit-active {
  opacity: 0.01;
  transition: opacity 500ms ease-in;
}