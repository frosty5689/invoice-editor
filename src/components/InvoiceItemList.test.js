import React from 'react';
import { InvoiceItemList } from './InvoiceItemList';
import renderer from 'react-test-renderer';

jest.mock('./AddInvoiceItem', () => {
  const AddInvoiceItemMock = () => <div />;
  return AddInvoiceItemMock;
});

describe('InvoiceItemList component', () => {

  it('InvoiceItemList should render correctly', () => {
    const lineItems = [{
      id: 1,
      content: {
        name: 'Widget',
        quantity: 2,
        price: 10.22
      }
    }, {
        id: 2,
        content: {
          name: 'Cogs',
          quantity: 4,
          price: 2.22
        }
    }];

    const component = renderer.create(
      <InvoiceItemList lineItemss={lineItems} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});