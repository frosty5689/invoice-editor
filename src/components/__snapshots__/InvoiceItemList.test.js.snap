// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceItemList component InvoiceItemList should render correctly 1`] = `
<tbody>
  <tr>
    <td>
      <div />
    </td>
    <td />
    <td />
    <td />
    <td />
  </tr>
</tbody>
`;
